agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/Configuration.java

```diff
@@ -269,17 +302,26 @@ private void validateSpanProcessorIncludeExclude() {
         public String value;
         public String fromAttribute;
 
-        public void validate() {
+        public void validate() throws FriendlyException {
 
                 if (this.key == null || this.key.isEmpty()) {
-                    throw new ConfigurationException("User provided config has invalid action with empty key!!!");
+                    throw new FriendlyException("ApplicationInsights Java Agent failed to start.",
```

Comment:
just a thought, instead of passing "ApplicationInsights Java Agent failed to start" everywhere, which requires knowledge of how this code is used, it may be nicer to only provide that part of the message when performing the log (e.g. from the catch block where we know that we are failing the agent to start)